addons/rmsmartshape/shapes/shape.gd

```diff
@@ -1736,13 +1763,56 @@ func _build_edge_with_material(
 		for q in new_quads:
 			edge.quads.push_back(q)
 		i += next_point_delta
-
+	
+	# leftover final taper for the last sharp corner if required
+	if taper_sharp:
+		if sharp_taper_next != null and edge.quads[0].corner == SS2D_Quad.CORNER.NONE:
+			var taper := _taper_quad(sharp_taper_next, edge_material, texture_idx, true)
+			if taper != null:
+				taper.ignore_weld_next = true
+				edge.quads.push_back(taper)
+			else:
+				sharp_taper_next.ignore_weld_next = true
+		sharp_taper_next = null
+	
 	if edge_material_meta != null:
 		if edge_material_meta.weld:
 			_weld_quad_array(edge.quads, edge.wrap_around)
 
 	return edge
 
+func _taper_quad(quad: SS2D_Quad, edge_mat: SS2D_Material_Edge, tex_idx: int, facing_right: bool) -> SS2D_Quad:
+	var taper_texture: Texture2D = (
+		edge_mat.get_texture_taper_right(tex_idx)
+			if facing_right else
+		edge_mat.get_texture_taper_left(tex_idx))
+	if taper_texture != null:
+		var taper_size: Vector2 = taper_texture.get_size()
+		var fit: bool = absf(taper_size.x) <= quad.get_length_average()
+		if fit:
+			var taper_quad := quad.duplicate()
+			taper_quad.corner = 0
```

Comment:
Please, set to `SS2D_Quad.CORNER.NONE` instead of 0

Reply:
@mphe Should we change that property to typed enum instead of int?

Reply:
Yes, this is likely a remain from 3.x and should be a typed enum.